test/integration/loginconsent/server.go

```diff
@@ -0,0 +1,363 @@
+	log.Printf("handling request: %s", r.URL.String())
+
+	skipConsent, err := r.Cookie(skipConsentCookieName)
+	if err != nil {
+		log.Printf("failed to fetch cookie %s: %s", skipConsentCookieName, err.Error())
+
+		return
+	}
+
+	log.Printf("consent skip value %s", skipConsent.Value)
+
+	if skipConsent.Value == skipConsentTrue {
+		s.completeConsent(w, r, &ConsentConfigRequest{UserClaims: &UserClaims{}}, r.URL.Query().Get("consent_challenge"))
+	}
+
+	challenge := r.URL.Query().Get("consent_challenge")
+	if challenge == "" {
+		log.Printf("missing consent_challenge")
+
+		return
+	}
+
+	http.SetCookie(w, &http.Cookie{
+		Name:  consentChallengeCookieName,
+		Value: challenge,
+	})
+
+	_, err = w.Write([]byte("mock consent UI"))
+	if err != nil {
+		log.Printf("failed to write imaginary UI: %s", err.Error())
+
+		return
+	}
+
+	log.Printf("rendered mock consent UI in response to request %s", r.URL.String())
+}
+
+func (s *server) userAuthorizationHandler(w http.ResponseWriter, r *http.Request) {
+	cookie, err := r.Cookie(consentChallengeCookieName)
+	if err != nil {
+		log.Printf("failed to fetch cookie %s: %s", consentChallengeCookieName, err.Error())
+
+		return
+	}
+
+	request := &ConsentConfigRequest{}
+
+	err = json.NewDecoder(r.Body).Decode(request)
+	if err != nil {
+		log.Printf("failed to decode user consent config request: %s", err.Error())
+
+		return
+	}
+
+	s.completeConsent(w, r, request, cookie.Value)
+}
+
+func (s *server) completeConsent(w http.ResponseWriter, r *http.Request, request *ConsentConfigRequest, consentChallenge string) {
+	consentReq, _, err := s.hydra.GetConsentRequest(r.Context()).ConsentChallenge(consentChallenge).Execute()
+	if err != nil {
+		log.Printf("failed to get hydra consent request: %s", err.Error())
+
+		return
+	}
+
+	b, err := s.store.Get(consentReq.GetSubject())
+	if err != nil {
+		log.Printf("failed to fetch test config for sub=%s: %s", consentReq.GetSubject(), err.Error())
+
+		return
+	}
+
+	var test testConfig
+
+	err = json.Unmarshal(b, &test)
+	if err != nil {
+		log.Printf("failed to unmarshal user data %s: %s", b, err.Error())
+
+		return
+	}
+
+	if request.Fail {
+		completedReq, _, err := s.hydra.RejectConsentRequest(r.Context()).ConsentChallenge(consentChallenge).Execute()
+		if err != nil {
+			log.Printf("failed to reject consent request at hydra: %s", err.Error())
+
+			return
+		}
+
+		redirectURL := completedReq.GetRedirectTo()
+
+		http.Redirect(w, r, redirectURL, http.StatusFound)
+		log.Printf("user did not consent; redirected to %s", redirectURL)
+
+		return
+	}
+
+	now := time.Now()
+
+	completedReq, _, err := s.hydra.AcceptConsentRequest(r.Context()).AcceptConsentRequest(client.AcceptConsentRequest{
+		GrantAccessTokenAudience: consentReq.GetRequestedAccessTokenAudience(),
+		GrantScope:               consentReq.GetRequestedScope(),
+		HandledAt:                &now,
+		Session: &client.ConsentRequestSession{
+			//AccessToken: nil,
```

Comment:
Looks like you left a commented out line here - if intentional, then you can ignore my comment